CMakeLists.txt

```diff
@@ -133,6 +128,29 @@ if(GIT_EXECUTABLE)
         -P ${CMAKE_CURRENT_SOURCE_DIR}/cmake/GenerateVersionHeader.cmake)
 endif()
 
+add_compile_definitions(SPDLOG_FMT_EXTERNAL)
+add_compile_definitions(FMT_HEADER_ONLY)
+add_compile_definitions(GTEST_USE_STATIC_LIBS)
+
+find_package(nlohmann_json REQUIRED)
+find_package(spdlog_setup REQUIRED)
+find_package(spdlog REQUIRED)
+find_package(fmt REQUIRED)
+find_package(cpptoml REQUIRED)
+
+include_directories(${nlohmann_json_INCLUDE_DIR})
+include_directories(${spdlog_INCLUDE_DIR})
+include_directories(${spdlog_setup_INCLUDE_DIR})
+include_directories(${fmt_INCLUDE_DIR})
+include_directories(${cpptoml_INCLUDE_DIR})
```

Comment:
Prefer using `target_link_libraries` with the CMake targets provided by the package config files loaded using `find_package()`. They provide and propagate the necessary linked libraries, include dirs, defines and build flags via `INTERFACE_INCLUDE_DIRECTORIES` and similar properties applied to the targets.

Reply:
Also, unlike `target_include_directories`, the global `include_directories()` info won't get exported with the generated CMake config files that should eventually be installed by the package.

Reply:
done!